runtime/src/bank.rs

```diff
@@ -1110,6 +1115,45 @@ impl Bank {
         )
     }
 
+    fn collect_rent(
+        &self,
+        txs: &[Transaction],
+        iteration_order: Option<&[usize]>,
+        loaded_accounts: &mut [Result<TransactionLoadResult>],
+    ) {
+        let mut collected_rent = 0;
+        let mut seen_credit_only_account: HashSet<Pubkey> = HashSet::new();
+        for (tx, loaded_account) in
+            OrderedIterator::new(txs, iteration_order).zip(loaded_accounts.iter())
+        {
+            if loaded_account.is_err() {
```

Comment:
this makes me worry that an AccountNotFound error caused by a rent deduction won't be handled correctly.  need a test for this up front...  e.g. rent due is 2 lamports, but account has only 1 lamport in it, the load() should generate an ANF *and* collect 1 lamport

Reply:
the rent value in the vector should be correct, so you don't need to consider the load error, I think...

Reply:
I did this, because same was checked in `collect_fees`

Reply:
right, but in collect_fees(), we only charge the issuing account the fee if we are preparing to execute the instructions.  in the case of rent, collecting rent itself can cause a TX to fail, but the account **_still owed_** the rent...

thinking more about this, a load failure will keep the transaction from making it to the ledger, so validators won't know to charge the rent.

so: I think your code's correct

Reply:
we can collect rent from issuing accounts at some future time